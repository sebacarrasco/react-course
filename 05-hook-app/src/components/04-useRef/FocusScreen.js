import React, { useRef } from 'react'
import "../02-useEffect/effects.css"
export const FocusScreen = () => {

    // El useRef sirve en este caso para tener referencia a elemntos html
    const inputRef = useRef();

    const handleClick = () => {
        inputRef.current.select();
        // Sin el html habr√≠a que haber hecho document.querySelector...
    }

    return (
        <div>
            <h1> Focus Screen </h1>
            <hr/>

            <input
                ref={ inputRef }
                className="form-control"
                placeholder="Su nombre"
            />

            <button
                className="btn btn-outline-primary mt-2"
                onClick={ handleClick }
            >
                Focus
            </button>
        </div>
    )
}
