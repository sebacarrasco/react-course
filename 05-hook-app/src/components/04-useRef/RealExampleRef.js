import React, { useState } from 'react';
import "../02-useEffect/effects.css";
import { MultipleCustomHooks } from '../03-examples/MultipleCustomHooks';

// useRef es básicamente un setState pero que solo retorna el state
// y se puede modificar directamente. Además, al modificarlo no se
// vuelve a renderear.
// Para lo que más se ocupa es para tener referencias a elementos del DOM (No es este el caso)
// En esta ocasión se usa para prevenir cambiar el state de un componente que ya no
// está montado

export const RealExampleRef = () => {

    const [show, setShow] = useState(false);

    return (
        <div>
            <h1>Real example with Ref</h1>
            <hr/>

            { show && <MultipleCustomHooks /> }

            <button
                className="btn btn-primary mt-5"
                onClick={ () => setShow(!show) }
            >
                Show/Hide
            </button>
        </div>
    )
};
