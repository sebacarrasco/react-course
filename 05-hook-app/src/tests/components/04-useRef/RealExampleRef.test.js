import React from 'react';
import { shallow } from "enzyme";

import { RealExampleRef } from "../../../components/04-useRef/RealExampleRef";

describe('Tests for <RealExampleRef />', () => {
   
    const wrapper = shallow(<RealExampleRef />);

    test('should match the snapshot', () => {
        
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find("MultipleCustomHooks").exists()).toBe(false);

    });


    test('should show the <MultipleCustomHooks /> component', () => {

        wrapper.find("button").simulate("click");

        expect(wrapper.find("MultipleCustomHooks").exists()).toBe(true);

    });
    

});
