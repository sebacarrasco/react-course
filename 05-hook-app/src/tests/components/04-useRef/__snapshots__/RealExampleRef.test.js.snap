// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <RealExampleRef /> should match the snapshot 1`] = `
<div>
  <h1>
    Real example with Ref
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;
